### PR TITLE
Icon dialog picker has been added to formgenerator now

### PR DESCRIPTION
**Pull request by frimpongopoku**
Implemented fontawesome selector dialog 
Used to have to add icons as text like this
Screenshot 2022-04-19 at 13 45 30

Now its like this: 

https://user-images.githubusercontent.com/26961591/163985557-49c385c1-2e2c-4f5f-a9ab-d58d8fdbcdb0.mov

### CODE REVIEW

**Comment by frimpongopoku** [comment]
That huge +1300.... number there is just from changing the location of the fontawesome json file. 🤣 🤣 🤣

**Comment by BradHN1** [comment]
I like it.  But I want to talk about a couple things before merging it.
a) I'd like to include a larger set of icons than we used to ahve available, does this do it?
b) I'd like to have the list start with the most frequently used icons.

**Comment by frimpongopoku** [comment]
> a) I'd like to include a larger set of icons than we used to have available, does this do it?

No, this set of icons is the same as we used to have. I can look into a list that has more. 

And yes, the most frequently used icons feature can be added.

**Comment by frimpongopoku** [comment]
@BradHN1  I found a new and more modern font-awesome icon set which has 1.5+k icons. Our old one had 700-ish... so yeah, its better. The new one contains all the icons the old one had and some more 👯‍♂️  


I also added the recently used feature: 

https://user-images.githubusercontent.com/26961591/164216479-9fb10c1f-c9e2-4ad7-8c51-aa86b618bbbe.mov

File: app/containers/MassEnergizeSuperAdmin/ME  Tools/icon dialog/IconDialog.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    setSearched(found);
+  };
+
+  const addToRecents = (iconName) => {
+    const rem = recents.filter((ic) => ic !== iconName);
+    var content = [iconName, ...rem];
+    if (content.length > maximumNumberOfRecentItems)
+      content = content.slice(0, maximumNumberOfRecentItems);
+    setRecents(content);
+    localStorage.setItem(iconsStorageKey, JSON.stringify(content));
```
Comment (Davidjwkinghorn) [inline app/containers/MassEnergizeSuperAdmin/ME  Tools/icon dialog/IconDialog.js:60]: This means that frequently used icons would persist across sessions right? 

I'm also assuming we want recent icons to be per admin, so this is good.

Comment (BradHN1) [reply app/containers/MassEnergizeSuperAdmin/ME  Tools/icon dialog/IconDialog.js:60]: No.  Ideally we want the most common icons chosen for our community sites to be at the top of the list.  Not per admin.

Comment (BradHN1) [reply app/containers/MassEnergizeSuperAdmin/ME  Tools/icon dialog/IconDialog.js:60]: We could do it manually, updating a json file or something that the front end can read in to list commonly used icons at the top.

**Pull request merged**